bofire/domain/constraint.py

```diff
@@ -231,17 +264,40 @@ class NonlinearConstraint(Constraint):
 
     type: Literal["NonlinearConstraint"] = "NonlinearConstraint"
     expression: str
+    jacobian_expression: str = None
+    features: FeatureKeys = None
```

Comment:
@bertiqwerty, we could add the features then also as required attribute for all constraints in the `Constraint` base class. I would prefer when it is always available.